### DISCUSSION TITLE
I can't find solution for days, Nextjs and Nodejs Together problem

### DISCUSSION THREAD

**Discussion by muhammedalibilgin** [vercel/next.js#53927] [Help] [answered]
### Summary

server1.js and server2.js is the same but server1.js is not work.
Here is my repository example. pls check readme file
https://github.com/muhammedalibilgin/nextjsAndNodejs

### Additional information

```js
i created "create-next-app" and then server2.js create into this.
"node server2" is succes no problem but server1.js is not work.
server1.js == server2.js same.
just diffrent between add "dir" in the " const app = next({ dev, dir: "./client2" }); "
```

### Example

https://github.com/muhammedalibilgin/nextjsAndNodejs.git

**Comment by devjiwonchoi** [comment]
`dir` must be an absolute path, try `path.resolve('client2')`

Example:

```js
const path = require("path");
const express = require("express");
const next = require("next");

const dev = process.env.NODE_ENV !== "production";
const app = next({ dev, dir: path.resolve('client2') });
const handle = app.getRequestHandler();

app.prepare().then(() => {
    const server = express();

    server.get("*", (req, res) => {
        return handle(req, res);
    });

    server.listen(3000, (err) => {
        if (err) throw err;
        console.log("> Ready on http://localhost:3000");
    });
});
```

**Reply by muhammedalibilgin** [reply]
Ok, i hope, i will try

**Reply by muhammedalibilgin** [reply]
i try it but i get the same error.
i would like to ask the question in a different way

server.js must be a higher direction

https://github.com/vercel/next.js/tree/canary/examples/custom-server

so like this:
+server.js
+client
--pages
--.next
--...

pls can you give me a working example?

**Comment by icyJoseph** [comment, marked as answer]
You need to fix how you access `express` and `next`, for example:

```js
const express = require("./client2/node_modules/express");
const next = require("./client2/node_modules/next");

const dev = process.env.NODE_ENV !== "production";
const app = next({ dev, dir: "./client2" });
const handle = app.getRequestHandler();

app.prepare().then(() => {
  const server = express();

  server.get("*", (req, res) => {
    return handle(req, res);
  });

  server.listen(3000, (err) => {
    if (err) throw err;
    console.log("> Ready on http://localhost:3000");
  });
});
```

Notice the first two lines. Since neither express, or next, are native Node packages, the Node.js module resolution algorithm won't find them. In this case I pointed to the express and next, contained within client2.

**Reply by muhammedalibilgin** [reply]
thank you for your reply. i m very thanksful. because I've been trying for days and I haven't been able to do it. thats good.
